devices/YarpRobotLoggerDevice/README.md

```diff
@@ -1,47 +1,38 @@
-### iCubGenova04 strain2 FT-IMU logger
+# YARPRobotLoggerDevice
 
-A logger device based on `YarpSensorBridge` and `matioCpp` to record dataset from strain2 sensor boards mounted on the iCubGenova04 robot.
+The **YARPRobotLoggerDevice** is a YARP device based on `YarpSensorBridge` and `matioCpp` to record dataset from a YARP-based robot.
 
 
 
-### Launching
+## :running: How to use the device
 
-- Launch yarprobotinterface on the robot with `icub_wbd_inertials.xml`
```

Comment:
I would leave `with `icub_wbd_inertials.xml`` as in simulation you need to launch `wbd` separately.

Reply:
Regarding this, we may write different subsubsection depending on whether you are in a simulation or on the real robot. For instance


- Depending on your configuration file and on the data you want to collect you should run the main `yarprobotinterface` with a different option, for instance:
  - **`iCubGazeboV3`** Given [the configuration file](https://github.com/ami-iit/bipedal-locomotion-framework/blob/master/devices/YarpRobotLoggerDevice/app/robots/iCubGazeboV3/yarp-robot-logger.xml#L31-L35) you should run [`whole-body-dynamics`](https://github.com/robotology/whole-body-estimators/tree/master/devices/wholeBodyDynamics) to estimate the external wrenches. This can be easily done by running the following command:  
    ```sh
    YARP_ROBOT_NAME=iCubGazeboV3 yarprobotinterface --config launch-wholebodydynamics.xml  
    ```
    In the case, you started gazebo with the real-time clock option (`gazebo -slibgazebo_yarp_clock.so`), please postpone `YARP_CLOCK=/clock` to the previous command as:
    ```sh
    YARP_CLOCK=/clock YARP_ROBOT_NAME=iCubGazeboV3 yarprobotinterface --config launch-wholebodydynamics.xml  
    ```
  - **`iCubGenova09`** Given [the configuration file](https://github.com/ami-iit/bipedal-locomotion-framework/blob/master/devices/YarpRobotLoggerDevice/app/robots/iCubGenova09/yarp-robot-logger.xml#L31-L35) you just need to run the main `yarprobotinterface` on the `icub-head` pc.
  - **`iCubGenova04`** Given [the configuration file](https://github.com/ami-iit/bipedal-locomotion-framework/blob/master/devices/YarpRobotLoggerDevice/app/robots/iCubGenova04/yarp-robot-logger.xml#L31-L35) you cannot use the default robot-`yarprobotinterface` on the head. Indeed the device will try to log also the data streamed by the [`root_imu`](https://github.com/ami-iit/bipedal-locomotion-framework/blob/f017cea7786048a8239f6434f6e58c7782906326/devices/YarpRobotLoggerDevice/app/robots/iCubGenova04/yarp-robot-logger.xml#L28) device. You can easily enable the `root_imu` device on the robot by running the following command on the `icub-head` pc:  
    ```sh
    yarprobotinterface --config icub_wbd_inertials.xml  
    ```

Reply:
@isorrentino @GiulioRomualdi Do you think it is worth addressing, and maintaining all those cases in this `README`? Because I think some of those can change frequently at this stage (e.g. to be precise, for `iCubGenova09` currently you should run the configuration file in the `ami-iit` fork)

Reply:
we can remove the part related to `iCubGenova09` since as you said it is going to change in a short period.

Reply:
I agree with you. I would just leave the two cases simulation and real robot.

Reply:
I have updated the file with the detailed procedure for `iCubGenova04` and `iCubGazeboV3` as you were suggesting